zk/watcher.go

```diff
@@ -67,6 +67,8 @@ func Connect(zkServers []string, prefix string, connectTimeout, sessionTimeout t
 		watchedNodes:   make(map[string]watchedNode),
 	}
 
+	// QUESTION: do we need to call reconnect here? I don't think it is necessary because we will call
+	// reconnect() inside w.run() in another goroutine immediately.
```

Comment:
On the first loop inside `run`, we wait for errors.

Reply:
I feel we do unnecessary zookeeper connects. We can check the state of the connection. If it is already connected, we don't need to connect again and close the old connection.